controller/userController.js

```diff
@@ -170,101 +155,118 @@ const retrieveById = (req, res) => {
+            newUser.salt = null;
+
+            return res.status(HttpStatus.OK).json({ token: token, user: newUser });
+        } else {
+            return res.status(HttpStatus.NOT_FOUND).json({
+                message: 'User not found.'
+            });
+        }
+    }).catch(err => {
+        return res.status(HttpStatus.INTERNAL_SERVER_ERROR).json({
+            message: ValidationUtil.buildErrorMessage(err, 'changePersonalData', 'user')
+        });
+    });
+};
+
+const changePassword = (req, res) => {
+    const loggedUser = req.decoded;
+
+    if (!ValidationUtil.isValidObjectId(loggedUser._id)) {
+        return res.status(HttpStatus.BAD_REQUEST).json({ message: 'Invalid ID' });
+    }
+
+    let user = {};
     if (req.body.passwordHash) {
         let retVal = getPasswordHash(req.body.passwordHash);
         user.salt = retVal.salt;
         user.passwordHash = retVal.passwordHash;
     }
 
-    User.findOneAndUpdate({ _id: userId }, user, { useFindAndModify: false, new: true, runValidators: true }).then(newUser => {
+    User.findOneAndUpdate({ _id: loggedUser._id }, user, { useFindAndModify: false, new: true, runValidators: true }).then(newUser => {
         if (newUser) {
+            // Generate new JWT
+            const token = jwt.sign(JSON.stringify(newUser), AppConfig.SECRET);
             newUser.passwordHash = null;
             newUser.salt = null;
-            return res.status(HttpStatus.OK).json(newUser);
+
+            return res.status(HttpStatus.OK).json({ token: token, user: newUser });
         } else {
             return res.status(HttpStatus.NOT_FOUND).json({
-                status: StatusEnum['ERROR'],
                 message: 'User not found.'
             });
         }
     }).catch(err => {
         return res.status(HttpStatus.INTERNAL_SERVER_ERROR).json({
-            status: StatusEnum['ERROR'],
-            message: ValidationUtil.buildErrorMessage(err, 'update', 'user')
+            message: ValidationUtil.buildErrorMessage(err, 'changePassword', 'user')
         });
     });
 };
 
-const remove = (req, res) => {
-    const userId = req.params.id;
-    const loggedUser = req.decoded;
+const changeActiveStatus = (req, res) => {
+    if (!req.params || !req.params.id || req.params.id.length != AppConfig.OBJECT_ID_LEN) {
+        return res.status(HttpStatus.BAD_REQUEST).json({ message: 'Invalid ID' });
+    }
 
-    if (userId.length != AppConfig.OBJECT_ID_LEN) {
-        return res.status(HttpStatus.BAD_REQUEST).json({
-            status: StatusEnum['ERROR'],
-            message: 'Invalid ID'
-        });
+    if (!req.body || !req.body.status) {
+        return res.status(HttpStatus.BAD_REQUEST).json({ message: 'Invalid status' });
     }
 
-    User.findOneAndUpdate({ _id: userId, companyID: loggedUser.companyID }, { active: 0 }, { useFindAndModify: false, new: true, runValidators: true }).then(user => {
+    const userId = req.params.id;
+    const status = req.body.status;
```

Comment:
Why combine both `params` and `body`? Why not just send everything as parameters, no need to complicate calls that are to be sent from front-end.